### ISSUE TITLE
Random mirror # doesn't compile with g++

### ISSUE THREAD

**Issue by m-ender** [C0deH4cker/Fission#1]
When compiling with

```
g++ --std=c++11 *.cpp -o Fission
```

I get

```
Mirror.cpp: In member function ‘virtual bool fsn::Mirror::onHit(fsn::Atom&)’:
Mirror.cpp:65:38: error: ‘arc4random’ was not declared in this scope
    atom.dir = (atom.dir + arc4random() % 3 + 3) & 3;
                                      ^
```

Would you mind using a more platform independent PRNG there?

**Comment by C0deH4cker** [comment]
I actually didn't think anyone was using Fission so I never bothered completing my changes. Since I now know that at least one person is using it, I'll go back and finish making changes (and fix this bug you reported, thanks!).

**Comment by m-ender** [comment]
It's a fun challenge to program in this. I wrote `atoi` and `itoa` if you're interested:

```
     ;
/  Y@<+@0'L/  [Y]_L
R? % \  @  /  $J
  ;A          /[{+++++++++L
   _          ; ;
   UR+++++++++>/
              O
```

and

```
     /"0"/    \
R'x[@%  //    /
     \$@{   / \
   @    \     <\
   $       ;  ;}+++++++++L
   %@       A {/
   M     \@<+@/
   ~       @
  ;Z       K Y 0'L
         \  !A   /
```

Is there a more convenient way to get an atom with mass 10 than `R+++++++++`?

**Comment by C0deH4cker** [comment]
I have had a massive set of local changes for a long time that I never pushed as they weren't finished. They somewhat change the way the language works so I was hesitant to do so. I think that I'll push those in a new branch while I finish working on them and let people see what they think of the new Fission. It modifies some components and adds others.

**Comment by C0deH4cker** [comment]
Just fixed this locally, will be in the next group of changes.

**Comment by m-ender** [comment]
Let me know when you've pushed that new branch. I'm currently working on a larger program, so I guess that would break a lot (but maybe simplify some things).

**Comment by C0deH4cker** [comment]
That is the goal, yes. Not much changes, but there are a few usability changes. I'm open to suggestions as well.

**Comment by m-ender** [comment]
Somewhat off topic for this discussion, but I wanted to show you the larger program I mentioned: http://codegolf.stackexchange.com/a/50858/8478. I'd think you find some people interested in esolang design in general in the CodeGolf.SE chat.

**Comment by C0deH4cker** [comment]
Wow that's amazing! I'll definitely spend some time figuring out how it works later. Seems like a pretty good test case since it (ab)uses almost every component in the language!

**Comment by m-ender** [comment]
I've expanded that post with a very extensive explanation that comes close to a Fission tutorial. ;)

I also found a 6-byte quine: http://codegolf.stackexchange.com/a/50968/8478

I'm really looking forward to the new version you mentioned. I've got a bunch of ideas to improve Fission, but I'd like to see what you had in mind before proposing them.

**Comment by m-ender** [comment]
FYI, [there are now at least two people using Fission](http://codegolf.stackexchange.com/a/50923/8478). ;)

**Comment by C0deH4cker** [comment]
I noticed! I have some ideas for modifications to Fission, but I'd like to run them by you first for feedback. If you're on Freenode, I created #Fission-lang (#Fission is already registered, but nobody is in the channel :/)

**Comment by m-ender** [comment]
I haven't used IRC in forever, but I found a web client. I'd definitely prefer [gitter](https://gitter.im/) or just SE's chat though. ;)

**Comment by C0deH4cker** [comment]
Well I tried gitter. I can say that at the very least their iOS client is a piece of crap as it doesn't let you do anything besides join one of the handful of rooms listed by default and your organizations. I see no way to create a room or even leave a room you've joined. The idea of it is cool but the implementation is way off.

Also on SE's iOS client I can't figure out how to use chat :/.

**Comment by m-ender** [comment]
Ah yeah, SE's mobile apps don't support chat (yet?). There's a separate
(unofficial) app ChatSEy though, which is alright. But I'm okay with
Freenode if the other options are a hassle for you.

**Comment by C0deH4cker** [comment]
Apparently ChatSEy isn't an app for iOS. If you have a better option, feel free to suggest it. For now I think IRC is the most manageable option at least when I'm not at my computer. I still have about 6 hours until I get back from work, so I'll be on then.

**Comment by m-ender** [comment]
No problem. I will likely be asleep by then, so just dump your ideas in IRC
and I'll have a read tomorrow morning.

**Comment by m-ender** [comment]
Looks like this isn't really working out. I've edited my GitHub profile to show my email address, so feel free to just email me directly... asynchronous communication works better across our timezones.

**Comment by C0deH4cker** [comment]
Fixed in d019e2265e006f74de80cd9e6e39d0950fd2349a
